### PR TITLE
SDIT-1617 prisoner search api

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderCharge.java [hunk 1]
```diff
@@ -79,7 +81,8 @@ public class OffenderCharge extends AuditableEntity {
     private OffenderBooking offenderBooking;
 
     @ManyToOne(optional = false, fetch = FetchType.LAZY)
-    @JoinColumn(name = "CASE_ID", nullable = false)
+    @JoinColumn(name = "CASE_ID")
+    @NotFound(action = NotFoundAction.IGNORE)
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderCharge.java:85]: Handle orphaned ID with missing OFFENDER_CASE record.

File: src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java [hunk 2]
```diff
@@ -150,7 +150,7 @@ public List<InmateBasicDetails> getBasicInmateDetailsForOffenders(final Set<Stri
                     .map(offender ->
                         offender.toBuilder()
                             .firstName(WordUtils.capitalizeFully(offender.getFirstName()))
-                            .middleName(WordUtils.capitalizeFully(offender.getMiddleName()))
+                            .middleName(WordUtils.capitalizeFully(offender.getMiddleName()).trim())
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:153]: Trim middle names as we'll be doing that with the new endpoint.

File: src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java [hunk 3]
```diff
@@ -143,16 +143,22 @@ public List<InmateBasicDetails> getBasicInmateDetailsForOffenders(final Set<Stri
...
-                        offender.toBuilder()
-                            .firstName(WordUtils.capitalizeFully(offender.getFirstName()))
-                            .middleName(WordUtils.capitalizeFully(offender.getMiddleName()))
-                            .lastName(WordUtils.capitalizeFully(offender.getLastName()))
-                            .build()
+                    .map(offender -> {
+                            var middleName = "";
+                            if (offender.getMiddleName() != null) {
+                                middleName = WordUtils.capitalizeFully(offender.getMiddleName()).trim();
+                            }
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:155]: Trim middle names as we'll be doing that with the new endpoint.

Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/service/InmateService.java:155]: bit naff having a mutable variable, might be better to just do
```
                                middleName = StringUtils.trim(WordUtils.capitalizeFully(offender.getMiddleName()));
```

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**